### PR TITLE
Add back `max_pending_accept_reset_streams`

### PR DESCRIPTION
**Pull request by allan2**
Adds back this function which was previously in hyper 0.14.

The original merge into hyper is
[here](https://github.com/hyperium/hyper/pull/3201/files/e8296d1c5bded88a508f1ca1f35791c30803182d)

This follows the original closely, with some minor wording improvements in the outer line docs.

Closes https://github.com/hyperium/hyper/issues/3461.

### CODE REVIEW

**Comment by allan2** [comment]
I updated the version to 0.1.2, possibly prematurely. This function is used by tonic so I'd like to have it for the tonic hyper 1.0 update :)

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "hyper-util"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (seanmonstar) [inline Cargo.toml:3]: We change the version at release time.

Comment (allan2) [reply Cargo.toml]: Version reverted

**Comment by allan2** [comment]
Hi @seanmonstar, I've synced and updated this PR.

With https://github.com/hyperium/hyper/pull/3507, this should complete the re-addition of this function.

**Comment by ikrivosheev** [comment]
https://github.com/hyperium/hyper-util/pull/90 was merged, we can close this PR.

**Comment by allan2** [comment]
Superseded by #90 (merged) and #102

**Pull request closed without merging**